### DISCUSSION TITLE
react-select integration

### DISCUSSION THREAD

**Discussion by elringus** [KingSora/OverlayScrollbars#533] [Show and tell]
I've integrated OverlayScrollbars with [react-select](https://github.com/jedwatson/react-select), which required some hacks to support scrolling to the focused option when navigating the list with arrow keys.

Hopefully it'll save someone a couple of hours: https://codesandbox.io/s/overlayscrollbars-react-select-dsq9q4?file=/index.js

I've initially attempted routing scroll viewport to the select component ([which works with another scroller](https://github.com/JedWatson/react-select/issues/4227#issuecomment-742848886)), but it doesn't seem to work with OverlayScrollbars . If anyone happen to find a more elegant solution, please share.

**Comment by KingSora** [comment]
@Elringus After a bit of digging (mostly understanding the `react-select` api) I came up with this solution: https://codesandbox.io/s/overlayscrollbars-react-select-forked-mjdmt2?file=/index.jsx which should get rid of all the custom scrolling logic :)

**Reply by elringus** [reply]
Brilliant, thanks a lot for sharing! The `innerRef={setViewport}` is what I've been missing. Didn't know you can set refs like that; still new to this react stuff :)
